android/src/main/java/com/onesignal/rnonesignalandroid/RNOneSignal.java

```diff
@@ -573,6 +575,20 @@ public void removeTags(ReadableArray tagKeys) {
         OneSignal.getUser().removeTags(RNUtils.convertReadableArrayIntoStringCollection(tagKeys));
     }
 
+    @ReactMethod
+    public void getTags(Promise promise) {
+        try {
+            Map<String, String> tags = OneSignal.getUser().getTags();
+            WritableMap writableTags = Arguments.createMap();
+            for (Map.Entry<String, String> entry : tags.entrySet()) {
+                writableTags.putString(entry.getKey(), entry.getValue());
+            }
+            promise.resolve(writableTags);
+        } catch (Throwable t) {
+            promise.reject(t.getMessage());
+        }
```

Comment:
Looking at this again, there is no need for the try-catch. We would need it if we are working with JSON and need to handle possible JSONExceptions, but we aren't doing so here. 
I did notice that we have a lot of redundant try-catches in this file that can potentially hide problems. May be worth cleaning up in a future refactor.